### PR TITLE
Remove the git-sync pebble service - only use `--one-time`

### PR DESCRIPTION
**Pull request by sed-i**
In this PR the git-sync pebble service is removed, in favor of `--one-time` only sync invocations. This is needed because of the following:
1. `git-sync --one-time` is needed by the `sync-now` action, but that interferes with the regular scheduled operation of `git-sync` - after the `sync-now` command, scheduled syncs do not work anymore. Related: 
2. Even if they worked together, there could be a race condition between git-sync updating the repo folder and charm code reading alert rules from disk.

The approach in this PR aligns `update-status` period with syncing period.

### CODE REVIEW

**Review by mmanciop — COMMENTED** [review]
So… now that we remove the asynchronous element of gitsync (and we do it for very good reasons), what does gitsync give us over a simple, plain usage of either the git command line, or a git client written in python?

**Comment by sed-i** [comment]
> So… now that we remove the asynchronous element of gitsync (and we do it for very good reasons), what does gitsync give us over a simple, plain usage of either the git command line, or a git client written in python?

It has a bunch of good features that we can avoid implementing ourselves:
- update symlink to repo dir if fetch is successful - useful for not wiping out the repo when there's no network
- hooks
- auth
- ssh
- in general git cli args acrobatics we could save some time on

**Review by simskij — APPROVED** [review]

File: src/charm.py [hunk 1]
```diff
@@ -143,98 +153,94 @@ def _common_exit_hook(self) -> None:  # noqa: C901
...
-                self.unit.status = BlockedStatus(str(e))
-                return
-
-        # Need to call this again in case files showed up on disk after the last hook fired.
-        self._update_relation_data()
+        self._update_hash_and_rel_data()
 
         if self._stored_hash in [self._hash_placeholder, None]:
-            self.unit.status = BlockedStatus("No hash file yet - wait for update-status")
+            self.unit.status = BlockedStatus("No hash file yet - confirm config is valid")
```
Comment (pengale) [inline src/charm.py:175]: +1 to this change, as this is now an appropriate use of the blocked status :-)

File: src/charm.py [hunk 2]
```diff
@@ -143,98 +153,94 @@ def _common_exit_hook(self) -> None:  # noqa: C901
...
+        # Go through the common exit hook to update the store hash
         self._common_exit_hook()
 
+    @property
+    def _configured(self) -> bool:
+        """Check if charm is in 'configured' state.
+
+        The charm is considered 'configured' if the `git_repo` config option is set.
+        """
+        return bool(self.config.get("git_repo"))
```
Comment (pengale) [inline src/charm.py:214]: I like this a lot, and think that it's a great pattern.

**Review by pengale — APPROVED** [review]
I am very much +1 on this. It removes a lot of brittle code, and replaces it with much more straightforward and understandable checks.

+1

**Pull request merged**